File: problem_9/pythagorean_triplet_test.rb
require 'minitest/autorun'
require 'minitest/pride'

class PythagoreanTripletTest < Minitest::Test
  def pythagorean_triplet(sum)
    (1..(sum/2)).to_a.reverse.each do |c|
      (1..(sum/6)).each do |i|
        b = c - i
        a = sum - c - b
        return [a, b, c] if a ** 2 + b ** 2 == c ** 2
      end
    end
  end

  def test_pythagorean_triplet
    assert_equal [3, 4, 5], pythagorean_triplet(12)
    assert_equal [5, 12, 13], pythagorean_triplet(30)
    assert_equal [80, 150, 170], pythagorean_triplet(400)
    assert_equal [200, 375, 425], pythagorean_triplet(1000)
    assert_equal [250, 600, 650], pythagorean_triplet(1500)
  end
end

File: problem_9/pythagorean-triplet.test.js
function pythagoreanTriplet(sum) {
  for (c = sum / 2; c > 0; c--) {
    for (i = 1; i < sum / 6; i++) {
      let b = c - i;
      let a = sum - b - c;
      if (Math.pow(a, 2) + Math.pow(b, 2) === Math.pow(c, 2)) {
        return [a, b, c];
      }
    }
  }
}

test("pythagoreanTriplet", () => {
  expect(pythagoreanTriplet(12)).toStrictEqual([3, 4, 5]);
  expect(pythagoreanTriplet(30)).toStrictEqual([5, 12, 13]);
  expect(pythagoreanTriplet(400)).toStrictEqual([80, 150, 170]);
  expect(pythagoreanTriplet(1000)).toStrictEqual([200, 375, 425]);
  expect(pythagoreanTriplet(1500)).toStrictEqual([250, 600, 650]);
});
